symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -45,46 +42,72 @@
-      StreamsApi streamsApi, PodApi podApi, AttachmentsApi attachmentsApi, DefaultApi defaultApi,
-      AuthSession authSession, RetryWithRecoveryBuilder retryBuilder) {
+  public MessageService(
+      MessagesApi messagesApi,
+      MessageApi messageApi,
+      MessageSuppressionApi messageSuppressionApi,
+      StreamsApi streamsApi,
+      PodApi podApi,
+      AttachmentsApi attachmentsApi,
+      DefaultApi defaultApi,
+      AuthSession authSession,
+      RetryWithRecoveryBuilder<?> retryBuilder
+  ) {
     this(messagesApi, messageApi, messageSuppressionApi, streamsApi, podApi, attachmentsApi, defaultApi, authSession,
         new TemplateResolver(), retryBuilder);
   }
 
-  public MessageService(MessagesApi messagesApi, MessageApi messageApi, MessageSuppressionApi messageSuppressionApi,
-      StreamsApi streamsApi, PodApi podApi, AttachmentsApi attachmentsApi, DefaultApi defaultApi,
-      AuthSession authSession, TemplateEngine templateEngine, RetryWithRecoveryBuilder retryBuilder) {
+  public MessageService(
+      MessagesApi messagesApi,
+      MessageApi messageApi,
+      MessageSuppressionApi messageSuppressionApi,
+      StreamsApi streamsApi,
+      PodApi podApi,
+      AttachmentsApi attachmentsApi,
+      DefaultApi defaultApi,
+      AuthSession authSession,
+      TemplateEngine templateEngine,
+      RetryWithRecoveryBuilder<?> retryBuilder
+  ) {
     this(messagesApi, messageApi, messageSuppressionApi, streamsApi, podApi, attachmentsApi, defaultApi, authSession,
         new TemplateResolver(templateEngine), retryBuilder);
   }
 
-  private MessageService(MessagesApi messagesApi, MessageApi messageApi, MessageSuppressionApi messageSuppressionApi,
-      StreamsApi streamsApi, PodApi podApi, AttachmentsApi attachmentsApi, DefaultApi defaultApi,
-      AuthSession authSession, TemplateResolver templateResolver, RetryWithRecoveryBuilder retryBuilder) {
-    this.messagesApi = messagesApi;
+  private MessageService(
+      MessagesApi messagesApi,
+      MessageApi messageApi,
+      MessageSuppressionApi messageSuppressionApi,
+      StreamsApi streamsApi,
+      PodApi podApi,
+      AttachmentsApi attachmentsApi,
+      DefaultApi defaultApi,
+      AuthSession authSession,
+      TemplateResolver templateResolver,
+      RetryWithRecoveryBuilder<?> retryBuilder
+  ) {
+    super(messagesApi, authSession, retryBuilder, templateResolver);
     this.messageApi = messageApi;
     this.messageSuppressionApi = messageSuppressionApi;
     this.streamsApi = streamsApi;
     this.podApi = podApi;
     this.attachmentsApi = attachmentsApi;
     this.authSession = authSession;
-    this.templateResolver = templateResolver;
     this.defaultApi = defaultApi;
-    this.retryBuilder = retryBuilder;
+  }
+
+  @Override
+  public OboMessageService obo(AuthSession oboSession) {
+    return new OboMessageService(this.messagesApi, oboSession, this.retryBuilder, templateResolver);
```

Comment:
new instance of `OboMessageService` is returned each time `obo(oboSession)` is called. I guess we’ll back to the memory problem.